lib/ansible/template/__init__.py

```diff
@@ -626,8 +627,13 @@ def _lookup(self, name, *args, **kwargs):
                 raise AnsibleUndefinedVariable(e)
             except Exception as e:
                 if self._fail_on_lookup_errors:
-                    raise AnsibleError("An unhandled exception occurred while running the lookup plugin '%s'. Error was a %s, "
-                                       "original message: %s" % (name, type(e), e))
+                    msg = "An unhandled exception occurred while running the lookup plugin '%s'. Error was a %s, original message: %s" % (name, type(e), e)
+                    if errors == 'warn':
+                        display.warning(msg)
+                    elif errors == 'ignore':
+                        display.log(msg)
+                    else:
+                        raise AnsibleError(msg)
```

Comment:
The way this is done is going to lead to unicode errors eventually.  This is probably the least messy way to handle that:

``` python
                    msg = u"An unhandled exception occurred while running the lookup plugin '%s'. Error was a %s, original message: %s" % (name, type(e), to_text(e)) 
                    if errors == 'warn':
                        display.warning(msg)
                    elif errors == 'ignore':
                        display.log(msg)
                    else:
                        raise AnsibleError(to_native(msg))
```

Reply:
updated